library/src/main/java/com/bumptech/glide/load/engine/cache/DiskLruCacheFactory.java

```diff
@@ -0,0 +1,68 @@
+package com.bumptech.glide.load.engine.cache;
+
+import java.io.File;
+
+/**
+ * Creates an {@link com.bumptech.glide.disklrucache.DiskLruCache} based disk cache in the specified disk cache
+ * directory.
+ * <p/>
+ * If you need to make I/O access before returning the cache directory use
+ * the {@link DiskLruCacheFactory#DiskLruCacheFactory(CacheDirectoryGetter, int)} constructor variant.
+ */
+public class DiskLruCacheFactory implements DiskCache.Factory {
+
+    private final int diskCacheSize;
+    private final CacheDirectoryGetter cacheDirectoryGetter;
+
+    /**
+     * Interface called out of UI thread to get the cache folder.
+     */
+    public interface CacheDirectoryGetter {
+        File getCacheDirectory();
+    }
+
+    public DiskLruCacheFactory(final String diskCacheFolder, int diskCacheSize) {
+        this(new CacheDirectoryGetter() {
+            @Override
+            public File getCacheDirectory() {
+                return new File(diskCacheFolder);
+            }
+        }, diskCacheSize);
+    }
+
+    public DiskLruCacheFactory(final String diskCacheFolder, final String diskCacheName, int diskCacheSize) {
+        this(new CacheDirectoryGetter() {
+            @Override
+            public File getCacheDirectory() {
+                return new File(diskCacheFolder, diskCacheName);
+            }
+        }, diskCacheSize);
+    }
+
+    /**
+     * When using this constructor {@link CacheDirectoryGetter#getCacheDirectory()} will be called out of UI thread,
+     * allowing to do I/O access without performance impacts.
+     *
+     * @param cacheDirectoryGetter Interface called out of UI thread to get the cache folder.
+     * @param diskCacheSize        Desired max bytes size for the LRU disk cache.
+     */
+    public DiskLruCacheFactory(CacheDirectoryGetter cacheDirectoryGetter, int diskCacheSize) {
+        this.diskCacheSize = diskCacheSize;
+        this.cacheDirectoryGetter = cacheDirectoryGetter;
```

Comment:
May worth throwing an NPE here for idiots :) instead of waiting for build to notice it. It will crash both ways, but at least in the ctor you'll have a proper stacktrace.